src/Tree/ParseTreeProperty.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Antlr\Antlr4\Runtime\Tree;
+
+use \SplObjectStorage;
+
+/**
+ * Associate a property with a parse tree node.
+ *
+ * Useful with parse tree listeners that need to associate a value with a
+ * particular tree node. For instance when specifying a return value for the
+ * listener event method that visited a particular node.
+ * You can associate any type of variable with a node.
+ *
+ * Example use:
+ *
+ *     use Antlr\Antlr4\Runtime\Tree\ParseTreeProperty;
+ *
+ *     // Declare a ParseTreeProperty-container for values in your listener.
+ *     private $values;
+ *
+ *     // Make a new container in the constructor of the listener.
+ *     $this->values = new ParseTreeProperty();
+ *
+ *     // Use it from any method in your listener.
+ *     // Use it to store or retrieve any value, associated to a specific node.
+ *     $this->values->put($node, $someValue);
+ *     $aValue = $this->values->get($node);
+ *     $this->values->removeFrom($node);
+ *
+ *
+ * This class is implemented in PHP as a wrapper around \SplObjectStorage.
+ */
+class ParseTreeProperty
+{
+	protected $storage;
```

Comment:
why not private?

Reply:
Yep. It was private first, but I changed it to protected to be abele to use it in the drived class... which is not a good idea. Change it back. Thanks.